src/v/raft/consensus.cc

```diff
@@ -2436,6 +2436,37 @@ void consensus::update_suppress_heartbeats(
     }
 }
 
+void consensus::update_heartbeat_status(vnode id, bool success) {
+    if (auto it = _fstats.find(id); it != _fstats.end()) {
+        if (success) {
+            it->second.heartbeats_failed = 0;
+        } else {
+            it->second.heartbeats_failed++;
+        }
+    }
+}
+
+bool consensus::should_reconnect_follower(vnode id) {
+    if (auto it = _fstats.find(id); it != _fstats.end()) {
+        auto last_at = it->second.last_hbeat_timestamp;
+        const auto fail_count = it->second.heartbeats_failed;
+
+        auto is_live = last_at + _jit.base_duration() > clock_type::now();
+        auto since = std::chrono::duration_cast<std::chrono::milliseconds>(
+                       clock_type::now() - last_at)
+                       .count();
+        vlog(
+          _ctxlog.trace,
+          "should_reconnect({}): {} fails, last ok {}ms ago",
+          id,
+          fail_count,
+          since);
+        return fail_count > 3 && !is_live;
```

Comment:
move the `3` a `static constexpr` global var we can change

Reply:
Done.  Thoughts on making it field-configurable?  I'm torn between wanting to be able to tweak it, and not wanting to have a million config options.